### DISCUSSION TITLE
Activate spindle modbus

### DISCUSSION THREAD

**Discussion by hanifhan** [grblHAL/core#310] [Q&A]
hi , i'd like to activate spindle modbus to capturing or get  data  ( encoder , rpm , freq ) from my VFD  (YX9000 series), is there any documentation or tutorial to activate and test it ? im using nucleo stm32f446re

**Comment by terjeio** [comment]
You will have to [write a driver](https://github.com/grblHAL/Plugins_spindle/tree/master/vfd) for it if you cannot configure and use the [modvfd driver](https://github.com/grblHAL/Plugins_spindle/tree/master#modvfd). If you want to read more data than supported by a driver you can add a plugin for that or write a more advanced driver.

**Comment by dresco** [comment]
Looks like it will probably work with ModBus RTU and the modvfd driver, your challenge may be getting the register definitions.. The vendor manual for the [YX9000](http://www.yuanxindrive.com/Upload/1471502097.pdf) says "see chapter 9", and then conveniently stops after chapter 8  :grin: 

The manual for another model ([YX3000](http://www.yuanxindrive.com/Upload/1470909799.pdf)) does include ModBus RTU register definitions though, so that would be my starting point. 

If I was doing it, I would hook it up to a PC with a USB-RS485 adapter, and use pymodbus to verify everything first.
